share/completions/git.fish

```diff
@@ -1426,10 +1426,22 @@ complete -f -c git -n '__fish_git_using_command reset; and not contains -- -- (c
 
 ### restore
 complete -f -c git -n '__fish_git_needs_command' -a restore -d 'Restore working tree files'
-complete -f -c git -n '__fish_git_using_command restore' -s s -l source -d 'Specify the source tree used to restore working tree'
+complete -f -c git -n '__fish_git_using_command restore' -s s -l 'source=' -d 'Specify the source tree used to restore the working tree'
 complete -f -c git -n '__fish_git_using_command restore' -s p -l patch -d 'Interactive mode'
-complete -f -c git -n '__fish_git_using_command restore' -s W -l worktree -d 'Restore working tree (default)'
+complete -f -c git -n '__fish_git_using_command restore' -s W -l worktree -d 'Restore the working tree (default)'
 complete -f -c git -n '__fish_git_using_command restore' -s S -l staged -d 'Restore the index'
+complete -f -c git -n '__fish_git_using_command restore' -s q -l quiet -d 'Suppress messages'
+complete -f -c git -n '__fish_git_using_command restore' -l progress -d 'Report progress status to standard error (default)'
+complete -f -c git -n '__fish_git_using_command restore' -l no-progress -d 'Do not report progress status to standard error'
+complete -f -c git -n '__fish_git_using_command restore' -l ours -d 'When restoring files, use stage #2 (ours)'
+complete -f -c git -n '__fish_git_using_command restore' -l theirs -d 'When restoring files, use stage #3 (theirs)'
+complete -f -c git -n '__fish_git_using_command restore' -s m -l merge -d 'Recreate the conflicted merge in the unmerged paths when restoring files'
+complete -f -c git -n '__fish_git_using_command restore' -l 'conflict=merge' -d 'Same as --merge, but specify \'merge\' as the style of the conflicting hunks (default)'
+complete -f -c git -n '__fish_git_using_command restore' -l 'conflict=diff3' -d 'Same as --merge, but specify \'diff3\' as the style of the conflicting hunks'
+complete -f -c git -n '__fish_git_using_command restore' -l ignore-unmerged -d 'When restoring files, do not abort the operation if there are unmerged entries'
+complete -f -c git -n '__fish_git_using_command restore' -l ignore-skip-worktree-bits -d 'Ignore the sparse patterns and unconditionally restore files in <pathspec>'
```

Comment:
What is a "sparse pattern"?

Git's documentation can sometimes be quite jargony, and we should try to improve on that.